File: sep2018/sep06/move_zeroes.rb
def move_zeroes(nums)
    front = 0
    
    i = 0
    while i < nums.length
        if nums[i] != 0
            nums[front] = nums[i]
            front += 1
        end
        
        i += 1
    end
    
    while front < nums.length
        nums[front] = 0
        
        front += 1
    end
    
    nums
end

File: sep2018/sep06/move_zeroes.js
var moveZeroes = function(nums) {
  for (let i = 0; i < nums.length; i++) {
    let current = nums[i];

    if (current === 0) {
      let j = i + 1;
      let swapIndex = null;
      while (!swapIndex && j < nums.length) {
        if (nums[j] !== 0) {
          swapIndex = j;
          break;
        }
        j++;
      }

      if (!swapIndex) {
        return;
      } else {
        [nums[i], nums[j]] = [nums[j], nums[i]];
      }
    }
  }
  console.log(nums);
};
